webrender/src/resource_cache.rs

```diff
@@ -321,14 +329,37 @@ impl ResourceCache {
         webgl_texture.size = size;
     }
 
-    pub fn request_image(&mut self,
-                         key: ImageKey,
-                         rendering: ImageRendering) {
+    pub fn request_image(&mut self, key: ImageKey, rendering: ImageRendering) {
         debug_assert!(self.state == State::AddResources);
-        self.pending_image_requests.push(ImageRequest {
+        let request = ImageRequest {
             key: key,
             rendering: rendering,
-        });
+        };
+
+        let template = self.image_templates.get(&key).unwrap();
+        if let ImageData::Vector(ref data) = template.data {
+            if let Some(ref mut renderer) = self.vector_image_renderer {
+                let same_epoch = self.cached_images.resources.get(&request)
+                    .map(|entry|{ entry.epoch == template.epoch})
```

Comment:
nit: I think this would probably be tidier written as a match { Some(..) => {} None => {} } statement.